### PR TITLE
Update vehicle anims for motorcycle

### PR DESCRIPTION
**Pull request by Deniel00**
**Pull Request Description**

Updated in vehicle anims for working motorcycles.
kép
_old anim_
kép

**Pull Request Checklist**:
* [x] Have you followed the guidelines in our Contributing document and Code of Conduct?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/project-error/npwd/pulls) for the same update/change?
* [x] Have you built and tested NPWD in-game after the relevant change?

### CODE REVIEW

**Comment by Mojito-Fivem** [comment]
image
With your changes the ped now sits forward in the car and they have to leave the car, pull the phone out and put it away again to return to the normal driving animation. The same issue occurs on motorcycles where the pen is stuck with their hand in the air.

**Comment by RockySouthpaw** [comment]
You'd need to change the dict/anim based off the class the player is in.

https://github.com/project-error/npwd/blob/8bec2f254dc57f18f9964df60538df0db2446124/resources/client/animations/animation.service.ts#L50

```ts
private handleCallAnimation(playerPed: number) {
    if (IsPedInAnyVehicle(playerPed, true)) {
      this.handleOnCallInVehicle(playerPed);
    } else {
      this.handleOnCallNormal(playerPed);
    }
  }

  private handleOpenAnimation(playerPed: number) {
    if (IsPedInAnyVehicle(playerPed, true)) {
      this.handleOpenVehicleAnim(playerPed);
    } else {
      this.handleOpenNormalAnim(playerPed);
    }
  }

  private handleCallEndAnimation(playerPed: number) {
    if (IsPedInAnyVehicle(playerPed, true)) {
      this.handleCallEndVehicleAnim(playerPed);
    } else {
      this.handleCallEndNormalAnim(playerPed);
    }
  }

  private handleCloseAnimation(playerPed: number) {
    if (IsPedInAnyVehicle(playerPed, true)) {
      this.handleCloseVehicleAnim(playerPed);
    } else {
      this.handleCloseNormalAnim(playerPed);
    }
  }
  ```
  You'd need to pass the class to the anim functions and then adjust the dics based off the class, can refer to the docs https://docs.fivem.net/natives/?_0x29439776AAA00A62. Your approach wouldn't work without rewriting your current pr entirely as you assume player vehicle = motorcycle where current method is just a blanket solution for most vehicle classes with the obvious exception of bikes. You'd just compare current class to see if its 8 and adjust the dics/anim locals accordingly.

**Comment by Deniel00** [comment]
As soon as possible I'm check this and editing.

**Comment by Deniel00** [comment]
I'm updated file for check player is in vehicle or motorcycles, but if in vehicles always restart opening animation and didn't find why made this.

File: resources/client/animations/animation.service.ts [hunk 1]
```diff
@@ -163,9 +167,13 @@ export class AnimationService {
     removePhoneProp();
   }
 
-  private async handleOnCallInVehicle(playerPed: number): Promise<void> {
-    const DICT = 'anim@cellphone@in_car@ps';
-    const ANIM = 'cellphone_call_listen_base';
+  private async handleOnCallInVehicle(playerPed: number, vehicleClass: number): Promise<void> {
+    var DICT = 'anim@cellphone@in_car@ps';
```
Comment (Mojito-Fivem) [inline resources/client/animations/animation.service.ts:171]: Can you use let/const syntax

File: resources/client/animations/animation.service.ts [hunk 2]
```diff
@@ -163,9 +167,13 @@ export class AnimationService {
...
+    var ANIM = 'cellphone_call_listen_base';
+		if (vehicleClass == 8){
```
Comment (Mojito-Fivem) [inline resources/client/animations/animation.service.ts:173]: formatting here

File: resources/client/animations/animation.service.ts [hunk 3]
```diff
@@ -182,10 +190,15 @@ export class AnimationService {
     }
   }
 
-  private async handleCallEndVehicleAnim(playerPed: number): Promise<void> {
-    const DICT = 'anim@cellphone@in_car@ps';
-    const ANIM = 'cellphone_call_to_text';
-    StopAnimTask(playerPed, DICT, 'cellphone_call_listen_base', 1.0);
+  private async handleCallEndVehicleAnim(playerPed: number, vehicleClass: number): Promise<void> {
+    var DICT = 'anim@cellphone@in_car@ps';
```
Comment (Mojito-Fivem) [inline resources/client/animations/animation.service.ts:194]: use let/const

File: resources/client/animations/animation.service.ts [hunk 4]
```diff
@@ -182,10 +190,15 @@ export class AnimationService {
...
+    var ANIM = 'cellphone_call_listen_base';
+		if (vehicleClass == 8){
```
Comment (Mojito-Fivem) [inline resources/client/animations/animation.service.ts:196]: formatting

File: resources/client/animations/animation.service.ts [hunk 5]
```diff
@@ -121,9 +121,13 @@ export class AnimationService {
     }
   }
 
-  private async handleOpenVehicleAnim(playerPed: number): Promise<void> {
-    const dict = 'anim@cellphone@in_car@ps';
-    const anim = 'cellphone_text_in';
+  private async handleOpenVehicleAnim(playerPed: number, vehicleClass: number): Promise<void> {
+    var dict = 'anim@cellphone@in_car@ps';
```
Comment (Mojito-Fivem) [inline resources/client/animations/animation.service.ts:125]: Use let/const syntax

**Review by Mojito-Fivem — CHANGES_REQUESTED** [review]

**Comment by Deniel00** [comment]
Fixed formating and using let/const

**Comment by itschip** [comment]
I'll test it this week.

**Review by Mojito-Fivem — CHANGES_REQUESTED** [review]

**Pull request closed without merging**